controller/workitemtype.go

```diff
@@ -16,25 +16,39 @@ const (
 // WorkitemtypeController implements the workitemtype resource.
 type WorkitemtypeController struct {
 	*goa.Controller
-	db application.DB
+	db     application.DB
+	config WorkItemControllerConfiguration
+}
+
+type WorkItemControllerConfiguration interface {
+	GetCacheControlWorkItemType() string
+}
+
+func getCacheControlWorkItemType(config WorkItemControllerConfiguration) app.CacheControlConfig {
+	return func() string {
+		return config.GetCacheControlWorkItemType()
+	}
 }
 
 // NewWorkitemtypeController creates a workitemtype controller.
-func NewWorkitemtypeController(service *goa.Service, db application.DB) *WorkitemtypeController {
+func NewWorkitemtypeController(service *goa.Service, db application.DB, config WorkItemControllerConfiguration) *WorkitemtypeController {
 	return &WorkitemtypeController{
 		Controller: service.NewController("WorkitemtypeController"),
 		db:         db,
+		config:     config,
 	}
 }
 
 // Show runs the show action.
 func (c *WorkitemtypeController) Show(ctx *app.ShowWorkitemtypeContext) error {
 	return application.Transactional(c.db, func(appl application.Application) error {
-		res, err := appl.WorkItemTypes().Load(ctx.Context, ctx.WitID)
+		result, err := appl.WorkItemTypes().Load(ctx.Context, ctx.WitID)
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-		return ctx.OK(res)
+		return ctx.Conditional(*result, getCacheControlWorkItemType(c.config), func() error {
```

Comment:
the wrapper method there can be replaced by c.config.GetCacheControlWorkItemType

config.GetCacheControlWorkItemType() is also a 'func() string'